tasks/image_size.js

```diff
@@ -7,53 +7,92 @@
 
 'use strict';
 
+var path = require('path');
+var chalk = require('chalk');
+var sizer = require('image-size');
+
 module.exports = function(grunt) {
-	var path = require('path');
-	var chalk = require('chalk');
-	var sizer = require('image-size');
-	var types = [
-		'.png',
-		'.gif',
-		'.bmp',
-		'.psd',
-		'.jpg',
-		'.tiff',
-		'.webp'
-	];
-
-	grunt.registerMultiTask('image_size', 'retrieve image saze information', function() {
-		var options = this.options({
-			configObject : ''
-		});
-
-		this.files.forEach(function (file) {
-			if ((!options.configObject || !options.configObject.length) && !file.dest) {
-				grunt.fail.warn('Neither options.configObject nor dest specified for this task.');
-				return;
-			}
-
-			var output = [];
-			file.src.forEach(function (src) {
-				if (grunt.file.exists(src) && grunt.file.isFile(src) && types.indexOf(path.extname(src).toLowerCase()) !== -1) {
-					var dimensions = sizer(src);
-					output.push({
-						name : src,
-						width : dimensions.width,
-						height : dimensions.height
-					});
-
-					grunt.verbose.writeln('dimensions of ' + src + ' width: ' + dimensions.width.toString() + ', height: ' + dimensions.height.toString());
-				}
-			});
-
-			if (options.configObject && options.configObject.length) {
-				grunt.config.set(options.configObject, output);
-			}
-
-			if (file.dest) {
-				grunt.file.write(file.dest, JSON.stringify(output));
-				grunt.log.writeln('File ' + chalk.cyan(file.dest) + ' created.');
-			}
-		});
-	});
+  grunt.registerMultiTask(
+    'image_size',
+    'Retrieve images size information',
+    function() {
+      var options = this.options({
+        processName: undefined,
+        processEntry: undefined,
+        processSizes: undefined,
+        configObject: undefined,
+        replacer: null,
+        space: 2
+      });
+
+      if (!this.files.length) {
+        return grunt.log.error('No files or `options.configObject` specified.');
+      }
+
+      var processedFiles = 0;
+
+      this.files.forEach(function(file) {
+        if (!file.src.length)
+          return grunt.log.error(
+            'No source files specified for ' + chalk.cyan(dest) + '.'
+          );
+
+        if (!file.dest) return grunt.log.error('No dest file specified.');
+
+        var sizes = [];
+
+        file.src.forEach(function(src) {
+          if (!grunt.file.isFile(src)) return;
+
+          var size = sizer(src);
```

Comment:
if src has non-supported format (or corrupted), sizer throws error, so it must be a try-catch block here

Reply:
Due to nature of Grunt, `dest` will be never incorrect or anything like that. But yeap, the file might end up being corrupt or of the unsupported extension.

Wrapping things into try-catch block sometimes makes code slower (since it prevents some important V8 optimization). But yeap, it seems to be the only solution here.

Generally, there are two ways:

1. Let it be like it is. The task will completely fail on fatal error. Well, not worst scenario.
2. Wrap it into try-catch.

   It makes sense only if we want to let the task run _till the end_, even if some of the files resulted in an error.

   Solution like this used, for instance, [here](https://github.com/ArmorDarks/grunt-gray-matter/blob/master/tasks/grayMatter.js)

   But I'm still questioning myself is it a reasonable thing to do, since it might be annoying in big projects. You have thousands of images, you run the task and only when it finished it says that few files resulted in an error. I wonder, wouldn't it be more reasonable to fail right away.

   And if we decide that failing right away is better — wrapping sizer into try-catch block doesn't add anything to it.

Reply:
> But I'm still questioning myself is it a reasonable thing to do, since it might be annoying in big projects
You right, but than we rely on to assumptions:
- grunt will catch all thrown errors
- image-size will throw always Error objects (and not just string, for example)
And I think, we can't. If in image-size code will be added this: ``throw 'some shit';``, grunt-image-size will be broken, because it end without error with this throw.

Reply:
Well, it isn't that simple with Grunt.

Just run few tests with errors, and you will see that during logging Grunt for some reason cuts the trace of the error: https://i.imgur.com/jVnyKQO.png

So, if the only purpose to provide a more meaningful error, by try-catching here we won't provide better context.

Reply:
No, the purpose  of try-catch here is to report at least some error. If you change ``throw new Error('test me');`` to ``throw 'test me';`` task will end up successfully.

Reply:
It strange, but if I put ``throw 'test';`` inside task, I've got this: https://i.imgur.com/y58mf3W.png

Reply:
Tested this and got the same result. More of it, if a naked throw is added, it magically makes pass `grunt image_size:noImage` without errors at all, which is definitely not right.

Seems like a bug or intentional strange design choice in Grunt. Normally scripts will fail on any throw. `Error` constructor just attaches to it trace object.

Well, that's not our issue, I'd say. I'd propose to consider that for future and maybe even open Grunt issue. But let's keep it out of the scope of this PR. If it will turn to be indeed an issue, we will add later try-catch as a workaround.

Reply:
Agree.